test/test_jit.py

```diff
@@ -4633,6 +4633,19 @@ def f():
             return ss1.pop() + ss2.pop()
         test_equality(f, lambda x: x)
 
+    @skipIfRocm
+    @unittest.skipIf(IS_WINDOWS, "TODO: Fix this test case")
+    def test_torchbind_take_as_arg(self):
+        def foo(stackstring):
+            # type: (torch.classes._TorchScriptTesting_StackString)
```

Comment:
It's going to be common for people to export their custom class types in their own module like 
```
MyClass = torch.classe.my_namespace.MyClass
```
I am pretty sure that this implementat will not work with that. Types should get resolved through the python environment, not through a separate side-channel lookup.

Reply:
Python doesn't close over names used in type expressions (either in py3 syntax or mypy syntax) and I'm not sure how to get those names to reliably show up in the python resolution callback. Any ideas?

Reply:
@zdevito updated to address this issue. lmk if there's anything else to consider